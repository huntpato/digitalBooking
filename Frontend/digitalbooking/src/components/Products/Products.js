import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { NotFound } from './NotFound/NotFound'
import useFetch from "use-http";
import { ReCard } from "../ReCard/ReCard";
import styles from "./Products.module.css";

export const Products = () => {
  const place = localStorage.getItem("place");
  const date = localStorage.getItem("date");

  let startDate = date!== "," && new Date(date.split(",")[0]).toISOString().split("T")[0];
  let endDate =  date!== "," && new Date(date.split(",")[1]).toISOString().split("T")[0];
  let city= '';
 

  const setPlaceFormat = () => {
    if (place !== "false" && place !== null) {
      let splitPlace = place.replace(/,[^,]+$/, "");
      city = splitPlace.replace(/ /g, "%20");
    } 
  };
  setPlaceFormat();


  let url2= `https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/reservations/search/${city}?startdate=${startDate}&enddate=${endDate}`;

 
  const options = {
    headers: {
      "X-Requested-With": "XMLHttpRequest",
    },
  };


  let url = date!== ","
    ? url2
    : "https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/products";
  let searchDate = "";

  const setPlace = () => {
    if (url !== url2 && place !== "false" && place !== null) {
      url = `${url}/search?city=${city}`;
  }};
  setPlace();


  const setDate = () => {
    if (date) {
      let dates = date.split(/(\s+)/);
      searchDate = `${dates[4]} de ${dates[2]} del ${dates[6]} hasta el ${dates[22]} de ${dates[20]} del ${dates[24]}  `;
    }
  };
  setDate();

  const { loading, data = [] } = useFetch(url, options, []);
  console.log(data);
  const [products, setProducts] = useState(4);
  const {
    productsList,
    recommendation,
    recommendationCards,
    recommendationtitle,
    recommendationContainer,
    loadMore,
    title,
    placeInfo,
    titleContainer
  } = styles;
  let history = useHistory();

  const handleClick = () => {
    setProducts((prevProducts) => prevProducts + 4);
  };

 
  return (
    <div className={productsList}>
      <section className={title}>
        <div className={titleContainer}>
          <h4 className={placeInfo}>
            {" "}
            {place !== "false"
              ? `Resultados de tu búsqueda en ${place}`
              : `Sugerencias para tus próximas vacaciones`}
          </h4>{" "}
          <i className="fas fa-arrow-left" onClick={() => history.goBack()}></i>
        </div>
      </section>
      <div className={recommendationContainer}>
        <section className={recommendation}>
          <h4 className={recommendationtitle}>
            {date === ","
              ? `Productos disponibles`
              : `Productos disponibles para el día ${searchDate}`}
          </h4>
          <div className={recommendationCards}>
            {loading ? <p>Cargando Recomendaciones...</p> :
            data.length>=1 ? data.slice(0, products).map((product, index) => (
              <ReCard
                key={index}
                type={product.productCategory.name}
                title={product.productName}
                img={product.photoGallery[0]}
                alt={"Isla Fanta sia"}
                text={product.productDescription}
                loc={product.location}
                rate={product.review}
                id={product.id}
                services={product.services}
                latlong={product.coordinates}
              /> 
            )) : <NotFound />}
          </div>
          {data.lenght > 6 && (
            <button className={loadMore} onClick={handleClick}>
              Ver Mas
            </button>
          )}
        </section>
      </div>
      
    </div>
  );
};