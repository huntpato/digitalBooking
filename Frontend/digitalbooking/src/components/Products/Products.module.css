.productsList{
  min-height: 93vh;
  background-color: var(--background-color2);
  padding: 68px 0px 0px 0px;
}

.recommendationContainer{
  background-color: var(--background-color2);
  display: flex;
  justify-content: center;
  width:100%;
}

.recommendation {
  background-color: var(--background-color2);
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: space-around;
  padding: 30px 0px 20px 0px;
  width: 90vw;
}

.title{
  align-items: center;
  background-color: var(--color-typo);
  display:flex;
  flex-direction: column;
  justify-content: center;
  padding: 30px 0px 0px 0px;
}

.titleContainer{
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 20px 0px 10px 0px;
  margin-bottom: 15px;
  width: 90vw;
}

.title i{
  color:var(--background-white);
  padding: 0px;
}

.placeInfo{
  color: var(--background-white);
  font-size: 16px;
  width:90%
}

.recommendationtitle {
  font-size: 14px;
  margin-bottom: 10px;
}

.recommendationCards {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
}

.loadMore{
  align-items: center;
  background-color: var(--background-white);
  border: 1.4px solid var(--principal-color);
  border-radius: 5px;
  color: var(--principal-color);
  display: flex;
  font-weight: 700;
  justify-content: center;
  margin: 50px 450px 0 450px;
  padding: 6px 30px 5px 30px;
  text-align: center;
  transition: 0.5s;
}

.loadMore:active {
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
  transform: scale(0.97);
}

.loadMore:hover {
  background-color: var(--principal-color);
  color: var(--background-white);
} 

@media (max-width: 768px) {
  
  .recommendationCards {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }

}

@media (max-width: 480px) {
}
