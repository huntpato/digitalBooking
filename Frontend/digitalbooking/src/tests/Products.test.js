import React from "react";
import { shallow } from "enzyme";
import { Products } from "../components/Products/Products";
import "@testing-library/jest-dom";
import { expect } from "@jest/globals";

jest.mock("react-router-dom", () => ({
  useParams: jest.fn().mockReturnValue({ nifUuid: "nif123" }),
  useHistory: jest.fn(),
  useLocation: jest.fn().mockReturnValue({
    pathname: "/register",
    search: "",
    hash: "",
    state: null,
    key: "5nvxpbdafa",
  }),
}));
const date =
  "Sun Nov 28 2021 00:00:00 GMT-0300 (hora estándar de Argentina),Tue Nov 30 2021 00:00:00 GMT-0300 (hora estándar de Argentina";
const place = "Buenos Aires, Capital Federal, Argentina";

const localStorageMock = (function () {
  
  let store = {};

  return {
    getItem(key) {
      return store[key];
    },

    setItem(key, value) {
      store[key] = value;
    },

    clear() {
      store = {};
    },

    removeItem(key) {
      delete store[key];
    },

    getAll() {},
  };
})();

Object.defineProperty(window, "localStorage", { value: localStorageMock });

describe("Pruebas en componente  Products />", () => {
  const jsonId = "date";
  const jsonId2 = "place"
  const newJson = { date: date };
  const newJson2 = { place : place };
  window.localStorage.setItem(jsonId, JSON.stringify(newJson));
  window.localStorage.setItem(jsonId2, JSON.stringify(newJson2));

  const wrapper = shallow(<Products />);

  test("Comprobar que  Products /> renderiza correctamente.", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Verificar que el componente carga la clase de CSS correctamente", () => {
    const section = wrapper.find("section").at(0);
    expect(section.hasClass("title")).toBe(true);
  });

  test("Verificar que el titulo carga correctamente", () => {
    const section = wrapper.find(".placeInfo");
    expect(section).toBeDefined();
  });

  test("Verificar que el subtitulo carga correctamente", () => {
    const section = wrapper.find(".recommendationtitle");
    expect(section).toBeDefined();
  });
});
