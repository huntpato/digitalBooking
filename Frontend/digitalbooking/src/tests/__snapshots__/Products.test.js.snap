// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente  Products /> Comprobar que  Products /> renderiza correctamente. 1`] = `
<div
  className="productsList"
>
  <section
    className="title"
  >
    <div
      className="titleContainer"
    >
      <h4
        className="placeInfo"
      >
         
        Resultados de tu búsqueda en {"place":"Buenos Aires, Capital Federal, Argentina"}
      </h4>
       
      <i
        className="fas fa-arrow-left"
        onClick={[Function]}
      />
    </div>
  </section>
  <div
    className="recommendationContainer"
  >
    <section
      className="recommendation"
    >
      <h4
        className="recommendationtitle"
      >
        Productos disponibles para el día 28 de Nov del 2021 hasta el 30 de Nov del 2021  
      </h4>
      <div
        className="recommendationCards"
      >
        <p>
          Cargando Recomendaciones...
        </p>
      </div>
    </section>
  </div>
</div>
`;
